rotkehlchen/tests/unit/test_general_cache.py

```diff
@@ -0,0 +1,104 @@
+
+    def mock_call_contract(contract, ethereum, method_name, **kwargs):
+        if method_name == 'pool_count':
+            return 2  # if we don't limit pools count, the test will run for too long
+        return ethereum.call_contract(
+            contract_address=contract.address,
+            abi=contract.abi,
+            method_name=method_name,
+            **kwargs,
+        )
+
+    call_contract_patch = patch(
+        target='rotkehlchen.chain.ethereum.contracts.EthereumContract.call',
+        new=mock_call_contract,
+    )
+
+    with call_contract_patch:
+        rotkehlchen_instance.update_curve_pools_cache()
+
+    expected_lp_tokens_to_pools = {
+        # first 2 are registry pools
+        '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490': '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7',
+        '0xFd2a8fA60Abd58Efe3EeE34dd494cD491dC14900': '0xDeBF20617708857ebe4F679508E7b7863a8A8EeE',
+        # second 2 are metapool factory pools
+        '0x1F71f05CF491595652378Fe94B7820344A551B8E': '0x1F71f05CF491595652378Fe94B7820344A551B8E',
+        '0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890': '0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890',
+    }
+
+    expected_pool_coins = {
+        # first 2 are registry pools
+        '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7': '0x6B175474E89094C44Da98b954EedeAC495271d0F,0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,0xdAC17F958D2ee523a2206206994597C13D831ec7',  # noqa: E501
+        '0xDeBF20617708857ebe4F679508E7b7863a8A8EeE': '0x028171bCA77440897B824Ca71D1c56caC55b68A3,0xBcca60bB61934080951369a648Fb03DF4F96263C,0x3Ed3B47Dd13EC9a98b44e6204A523E766B225811',  # noqa: E501
+        # second 2 are metapool factory pools
+        '0x1F71f05CF491595652378Fe94B7820344A551B8E': '0x96E61422b6A9bA0e068B6c5ADd4fFaBC6a4aae27,0x57Ab1ec28D129707052df4dF418D58a2D46d5f51',  # noqa: E501
+        '0xFD5dB7463a3aB53fD211b4af195c5BCCC1A03890': '0xC581b735A1688071A1746c968e0798D642EDE491,0xD71eCFF9342A5Ced620049e616c5035F1dB98620',  # noqa: E501
+    }
+
+    lp_tokens_to_pools_in_cache = {}
+    pool_coins_in_cache = {}
+
+    lp_tokens_in_cache = GlobalDBHandler.get_cache_keys(cache_type=GeneralCacheType.CURVE_POOL_ADDR)  # noqa: E501
+    for lp_token_addr in lp_tokens_in_cache:
+        pool_addr = GlobalDBHandler.get_general_cache_value(
+            cache_type=GeneralCacheType.CURVE_POOL_ADDR,
+            name=lp_token_addr,
+        )
+        lp_tokens_to_pools_in_cache[lp_token_addr] = pool_addr
+
+        pool_coins = GlobalDBHandler.get_general_cache_value(
+            cache_type=GeneralCacheType.CURVE_POOL_TOKENS,
+            name=pool_addr,
+        )
+        pool_coins_in_cache[pool_addr] = pool_coins
+
+    assert lp_tokens_to_pools_in_cache == expected_lp_tokens_to_pools
+    assert pool_coins_in_cache == expected_pool_coins
+
+    # Check that the token was created
+    token = GlobalDBHandler().get_ethereum_token('0xD71eCFF9342A5Ced620049e616c5035F1dB98620')
+    assert token is not None
```

Comment:
Maybe check the actual token name is what is expected?
Same for the below

Reply:
Yeah, makes sense